.github/workflows/ci.yml

```diff
@@ -0,0 +1,74 @@
+# Find full documentation here https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions
+name: CI
+
+on:
+    pull_request:
+
+    # Manual invocation.
+    workflow_dispatch:
+
+    push:
+        branches:
+            - main
+jobs:
+    CI:
+        runs-on: ubuntu-latest
+        permissions:
+            # required by aws-actions/configure-aws-credentials
+            id-token: write
+            contents: read
+        steps:
+            - uses: actions/checkout@v3
+
+            - uses: aws-actions/configure-aws-credentials@v2
+              with:
+                  role-to-assume: ${{ secrets.GU_RIFF_RAFF_ROLE_ARN }}
+                  aws-region: eu-west-1
+
+            - uses: actions/setup-node@v3
+              with:
+                  node-version-file: '.nvmrc'
+                  cache: npm
+                  cache-dependency-path: 'package-lock.json'
+
+            - name: Setup Java
+              uses: actions/setup-java@v3
+              with:
+                  java-version: '8'
+                  distribution: 'corretto'
```

Comment:
Adding `cache: 'sbt'` here is apparently helpful for performance, I learnt [here](https://github.com/guardian/grid/pull/4146#discussion_r1346060792).